lynxkite-app/web/src/workspace/Workspace.tsx

```diff
@@ -128,24 +128,32 @@ function LynxKiteFlow() {
           !Number.isNaN(ch.position.y)
         ) {
           getYjsDoc(state).transact(() => {
-            Object.assign(node.position, ch.position);
+            node.position.x = ch.position.x;
+            node.position.y = ch.position.y;
           });
         } else if (ch.type === "select") {
         } else if (ch.type === "dimensions") {
-          getYjsDoc(state).transact(() => Object.assign(node, ch.dimensions));
+          getYjsDoc(state).transact(() => {
+            node.width = ch.dimensions.width;
+            node.height = ch.dimensions.height;
+          });
         } else if (ch.type === "remove") {
           wnodes.splice(nodeIndex, 1);
         } else if (ch.type === "replace") {
           // Ideally we would only update the parameter that changed. But ReactFlow does not give us that detail.
-          const u = {
-            collapsed: ch.item.data.collapsed,
-            // The "..." expansion on a Y.map returns an empty object. Copying with fromEntries/entries instead.
-            params: {
-              ...Object.fromEntries(Object.entries(ch.item.data.params)),
-            },
-            __execution_delay: ch.item.data.__execution_delay,
-          };
-          getYjsDoc(state).transact(() => Object.assign(node.data, u));
+          getYjsDoc(state).transact(() => {
+            if (node.data.collapsed !== ch.item.data.collapsed) {
+              node.data.collapsed = ch.item.data.collapsed;
+            }
+            if (node.data.__execution_delay !== ch.item.data.__execution_delay) {
+              node.data.__execution_delay = ch.item.data.__execution_delay;
+            }
+            for (const [key, value] of Object.entries(ch.item.data.params)) {
+              if (node.data.params[key] !== value) {
```

Comment:
It's this line! When closing/opening the box, the `collapsed` attribute is changed. But without this line we replace all the parameters.

I think I misunderstood the issue! Maybe replacing a CRDT object is okay (after all, that happens when you change a parameter). But you can't replace it with an object that's already in the CRDT document tree. So `params.foo = { bar }` is okay, but `params.foo = params.foo` is _not_ okay.

Still a bit of a mystery why we haven't seen this elsewhere.